.github/workflows/publish_to_maven_local.yml

```diff
@@ -4,12 +4,14 @@ on:
   push:
     branches:
       - main
+  pull_request:
   workflow_dispatch:
 
 jobs:
   publish:
     name: Publish pulumi-kotlin to Maven Local Repository
-    runs-on: [self-hosted, active]
+    runs-on: [ self-hosted, active ]
+    if: ${{ startsWith(github.head_ref, 'prepare-release') || github.event_name == 'workflow_dispatch' || github.event_name == 'push' }}
```

Comment:
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-the-head-or-base-branch-of-a-pull-request

> To run a job based on the pull request's head branch name (as opposed to the pull request's base branch name), use the github.head_ref context in a conditional.